updater/lib/dependabot/environment.rb

```diff
@@ -1,64 +1,96 @@
+require "sorbet-runtime"
+
 module Dependabot
   module Environment
+    extend T::Sig
+    extend T::Generic
+
+    sig { returns(String) }
     def self.job_id
-      @job_id ||= environment_variable("DEPENDABOT_JOB_ID")
+      @job_id ||= T.let(environment_variable("DEPENDABOT_JOB_ID"), T.nilable(String))
     end
 
+    sig { returns(String) }
     def self.job_token
-      @job_token ||= environment_variable("DEPENDABOT_JOB_TOKEN")
+      @job_token ||= T.let(environment_variable("DEPENDABOT_JOB_TOKEN"), T.nilable(String))
     end
 
+    sig { returns(T::Boolean) }
     def self.debug_enabled?
-      @debug_enabled ||= job_debug_enabled? || environment_debug_enabled?
+      @debug_enabled ||= T.let(job_debug_enabled? || environment_debug_enabled?, T.nilable(T::Boolean))
     end
 
+    sig { returns(Symbol) }
     def self.log_level
       debug_enabled? ? :debug : :info
     end
 
+    sig { returns(String) }
     def self.api_url
-      @api_url ||= environment_variable("DEPENDABOT_API_URL", "http://localhost:3001")
+      @api_url ||= T.let(environment_variable("DEPENDABOT_API_URL", "http://localhost:3001"), T.nilable(String))
     end
 
+    sig { returns(String) }
     def self.job_path
-      @job_path ||= environment_variable("DEPENDABOT_JOB_PATH")
+      @job_path ||= T.let(environment_variable("DEPENDABOT_JOB_PATH"), T.nilable(String))
     end
 
+    sig { returns(String) }
     def self.output_path
-      @output_path ||= environment_variable("DEPENDABOT_OUTPUT_PATH")
+      @output_path ||= T.let(environment_variable("DEPENDABOT_OUTPUT_PATH"), T.nilable(String))
     end
 
+    sig { returns(T.nilable(String)) }
     def self.repo_contents_path
-      @repo_contents_path ||= environment_variable("DEPENDABOT_REPO_CONTENTS_PATH", nil)
+      @repo_contents_path ||= T.let(environment_variable("DEPENDABOT_REPO_CONTENTS_PATH", nil), T.nilable(String))
     end
 
+    sig { returns(T.any(String, T::Boolean)) }
     def self.github_actions?
-      @github_actions ||= environment_variable("GITHUB_ACTIONS", false)
+      @github_actions ||= T.let(
+        environment_variable("GITHUB_ACTIONS", false),
+        T.nilable(T.any(String, T::Boolean))
+      )
     end
 
+    sig { returns(T.any(String, T::Boolean)) }
     def self.deterministic_updates?
-      @deterministic_updates ||= environment_variable("UPDATER_DETERMINISTIC", false)
+      @deterministic_updates ||= T.let(
+        environment_variable("UPDATER_DETERMINISTIC", false),
+        T.nilable(T.any(String, T::Boolean))
+      )
     end
 
+    sig { returns(T::Hash[String, T.untyped]) }
     def self.job_definition
-      @job_definition ||= JSON.parse(File.read(job_path))
+      @job_definition ||= T.let(JSON.parse(File.read(job_path)), T.nilable(T::Hash[String, T.untyped]))
     end
 
+    sig do
+      type_parameters(:T)
+        .params(variable_name: String, default: T.any(Symbol, T.type_parameter(:T)))
+        .returns(T.any(String, T.type_parameter(:T)))
+    end
```

Comment:
This is pretty hairy, what do you think about making `default` a String, the default value `"_undefined"` then in the 2 cases we return a Boolean above we do `== "false"`?

Reply:
Unfortunately Sorbet won't type narrow with a string of "_undefined". I think I will split out boolean values to a separate "environment_variable_boolean" method, which should simplify things a bit.

I know the type signature is a little complex, but it shouldn't need to be edited. It would be nicer if I could use the `:undefined` literal symbol, [like TypeScript][1], but [Sorbet doesn't support that][2].

[1]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types
[2]: https://github.com/sorbet/sorbet/issues/2504